Content.Client/SimpleStation14/Examine/CharacterInformation/Systems/CharacterInformationSystem.cs

```diff
@@ -0,0 +1,152 @@
+            var departments = _prototype.EnumeratePrototypes<DepartmentPrototype>().OrderBy(d => d.ID).Reverse();
+            var department = departments.FirstOrDefault(d => d.Roles.Contains(test));
+
+            if (department is not null)
+            {
+                // Department (ex: Command or Security)
+                var dept = string.Join(" ", Loc.GetString($"department-{department.ID}").Split(' ').Select(s => s[0].ToString().ToUpper() + s[1..].ToLower()));
+                // Redo the job title with the department color and department (ex: Captain (Command) or Security Officer (Security))
+                job = $"[color={department.Color.ToHex()}]{job} ({dept})[/color]";
+            }
+        }
+
+        // Get and set flavor text
+        if (_config.GetCVar(CCVars.FlavorText) &&
+            _entity.TryGetComponent<DetailExaminableComponent>(uid, out var detail))
+            flavorText = detail.Content;
+
+        _window.UpdateUi(uid, name, job, flavorText);
+        _window.Open();
+    }
+
+    /// <summary>
+    ///     Gets the ID card component from either a PDA or an ID card if the entity has one
+    /// </summary>
+    /// <param name="idUid">Entity to check</param>
+    /// <returns>ID card component if they have one on the entity</returns>
+    private IdCardComponent? GetId(EntityUid? idUid)
+    {
+        // PDA
+        if (_entity.TryGetComponent(idUid, out PdaComponent? pda) && pda.ContainedId is not null)
+            return _entity.GetComponent<IdCardComponent>(pda.ContainedId.Value);
+        // ID Card
+        if (_entity.TryGetComponent(idUid, out IdCardComponent? id))
+            return id;
+
+        return null;
+    }
+
+    /// <summary>
+    ///     Gets the name and job title from an ID card component
+    /// </summary>
+    /// <param name="uid">The entity to attempt information retrieval from</param>
+    /// <returns>Name, Job Title</returns>
+    private (string, string) GetNameAndJob(EntityUid uid)
+    {
+        string? name = null;
+
+        if (_entity.TryGetComponent<IdentityComponent>(uid, out var identity) &&
+            identity.IdentityEntitySlot.ContainedEntity != null)
+            name = _entity.GetComponent<MetaDataComponent>(identity.IdentityEntitySlot.ContainedEntity.Value).EntityName;
+
+        if (_inventory.TryGetSlotEntity(uid, "id", out var idUid))
+        {
+            var id = GetId(idUid);
+            if (id is not null)
+            {
+                name ??= id.FullName;
+                if (string.IsNullOrEmpty(name))
+                    name = "Unknown";
+
+                var jobTitle = id.JobTitle;
+                if (string.IsNullOrEmpty(jobTitle))
+                    jobTitle = "Unknown";
+                jobTitle = string.Join(" ", jobTitle.Split(' ').Select(s => s[0].ToString().ToUpper() + s[1..].ToLower()));
+
+                return (name, jobTitle);
+            }
+        }
+
+        return (name ?? "Unknown", "Unknown");
+    }
+}
```

Comment:
These 100% should not be here. Put some comments to make very clear that these should not be here, and it can live here for now until access stuff is handled in a not stupid way.